src/platforms/web/runtime/components/transition-group.js

```diff
@@ -175,7 +176,7 @@ function applyTranslation (c: VNode) {
   const newPos = c.data.newPos
   const dx = oldPos.left - newPos.left
   const dy = oldPos.top - newPos.top
-  if (dx || dy) {
+  if (!c.data.isDisplayNone && (dx || dy)) {
```

Comment:
Here you can check if `oldPos` is empty rect by comparing every field to `{ x: 0, y: 0, width: 0, height: 0, top: 0, right: 0, bottom: 0, left: 0 }`.